### PR TITLE
Creating new monitoring and alerting chapter (BSC#1138864)

### PR DESCRIPTION
**Pull request by asettle**
This PR is the beginning of the creation of the new
monitoring and alerting chapter. This chapter requires
further cleanup and investigation as per BSC#1112900.

As per the instructions in BSC#1138864, this PR will require
content from Jan on Grafana and Prometheus.

### CODE REVIEW

**Comment by asettle** [comment]
> This is substantial and I am not competent to verify its technical accuracy.
> 
> There were some new-to-me terms that I think should be in the glossary. I defer to others' judgement though. Suggestions:
> 
> * Prometheus
> * Grafana
> * Alertmanager
> * routing tree

As noted in the commit message, this is WIP. This is not new content at this time, this PR only moves pre-existing content into a self-containing chapter.

Good call on the glossary. I'll do that as part of this PR when I get the new content.

File: xml/admin_monitoring_alerting.xml [hunk 1]
```diff
@@ -0,0 +1,936 @@
...
+<!DOCTYPE chapter
+[
+  <!ENTITY % entities SYSTEM "entity-decl.ent">
+    %entities;
+]>
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.0"  xml:id="monitoring-alerting">
+ <title>Monitoring and Alerting</title>
+ <para>
+  By default, &deepsea; deploys a monitoring and alerting stack on the
+  &smaster;, allowing for multiple instances for a HA setup.
```
Comment (jan--f) [inline xml/admin_monitoring_alerting.xml]: In SES6 DeepSea no longer deploys monitoring by default, users have to explicitly define a prometheus role for prometheus and alertmanager and the grafana role for grafana.

File: xml/admin_monitoring_alerting.xml [hunk 2]
```diff
@@ -0,0 +1,936 @@
...
+  It consists of the following components:
```
Comment (jan--f) [inline xml/admin_monitoring_alerting.xml]: This list should probably list the alertmanager too?

File: xml/admin_monitoring_alerting.xml [hunk 3]
```diff
@@ -0,0 +1,936 @@
...
+    service running on all &sminion;s.
+   </para>
+  </listitem>
+ </itemizedlist>
+ <para>
+  The &prometheus; configuration and <emphasis>scrape</emphasis> targets
+  (exporting daemons) are setup automatically by &deepsea;. &deepsea; also
+  deploys a list of default alerts, for example <literal>health
+  error</literal>, <literal>10% OSDs down</literal>, or <literal>pgs
+  inactive</literal>.
```
Comment (jan--f) [inline xml/admin_monitoring_alerting.xml]: Add a section here with possible custom configuration as outlined here https://github.com/SUSE/DeepSea/wiki/New-Monitoring

**Comment by jan--f** [comment]
ok I added a few comments and updated the existing documentation in the deepsea repo. This is still not very succinct but I hope this makes some kind of sense. Any chance you can take a look at let me know what is unclear?

File: xml/admin_monitoring_alerting.xml (hunk 3 above)
Comment (asettle) [reply xml/admin_monitoring_alerting.xml]: Are you referring to: https://github.com/SUSE/DeepSea/wiki/New-Monitoring#pillar-variables ?

Comment (jan--f) [reply xml/admin_monitoring_alerting.xml]: spot on :+1:

File: xml/admin_monitoring_alerting.xml [hunk 4]
```diff
@@ -0,0 +1,1065 @@
...
+   the command line flags and a configuration file that defines inhibition
+   rules, notification routing and notification receivers.
+  </para>
+  <sect2>
+   <title>Configuration File</title>
+   <para>
+    &alertmanager;'s configuration is different for each deployment. Therefore,
+    &deepsea; does not ship any related defaults. You need to provide your own
+    <filename>&alertmanager;.yml</filename> configuration file. The
+    <package>&alertmanager;</package> package by default installs a
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
    <package>alertmanager</package> package by default installs a
```

File: xml/admin_monitoring_alerting.xml [hunk 5]
```diff
@@ -0,0 +1,1065 @@
...
+    configuration file <filename>/etc/prometheus/&alertmanager;.yml</filename>
```
Comment (tbazant) [inline xml/admin_monitoring_alerting.xml]:
```suggestion
    configuration file <filename>/etc/prometheus/alertmanager.yml</filename>
```

File: xml/admin_monitoring_alerting.xml [hunk 6]
```diff
@@ -0,0 +1,1065 @@
...
+  deploys a list of default alerts, for example <literal>health
+  error</literal>, <literal>10% OSDs down</literal>, or <literal>pgs
+  inactive</literal>.
+ </para>
+
+ <sect1 xml:id="pillar-variables">
+  <title>Pillar Variables</title>
+  <para>&spillar; contains information about a minion or many minions stored or
+    generated on the &smaster;. The pillar is already running in &salt; by
+    default. There are two main files:</para>
```
Comment (jan--f) [inline xml/admin_monitoring_alerting.xml]: Suggestion:
The pillar is a key-value store that provides information and configuration values to minions. It is availabel to all minions though potentially with different content. The pillar is pre-populated with default values and can be customized in two different ways:

File: xml/admin_monitoring_alerting.xml [hunk 7]
```diff
@@ -0,0 +1,1065 @@
...
+[
+  <!ENTITY % entities SYSTEM "entity-decl.ent">
+    %entities;
+]>
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xi="http://www.w3.org/2001/XInclude" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.0"  xml:id="monitoring-alerting">
+ <title>Monitoring and Alerting</title>
+ <para>
+  In &productname; 6, &deepsea; no longer deploys a monitoring and alerting
+  stack on the &smaster;. Users have to define the &prometheus; role
+  for &prometheus; and &alertmanager;, and the &grafana; role for &grafana;.
```
Comment (jan--f) [inline xml/admin_monitoring_alerting.xml]: Not sure if that was just in the PDF I reviewed but `In &productname; 6,` resulted in `In SUSE Enterprise Storage 6;,` and `alertmanager` should probably be capitalized, as it is a name.

Comment (asettle) [reply xml/admin_monitoring_alerting.xml]: Both of these are correct :) but &alertmanager; results in an uppercase solution: Alertmanager :)

**Pull request merged**